include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -490,6 +490,48 @@ namespace internal
         return {0, 0};
       }
 
+
+
+      // The next few internal helper functions are needed to support various
+      // DoFIndicesType kinds, e.g. actual vectors of DoFIndices or dummy
+      // types that we use when we only want to work on the internally stored
+      // DoFs and never extract any number.
+      static unsigned int
+      get_array_length(const std::vector<types::global_dof_index> &array)
+      {
+        return array.size();
+      }
+
+      static unsigned int
+      get_array_length(const int &)
+      {
+        return 0;
+      }
+
+      static types::global_dof_index &
+      get_array_entry(std::vector<types::global_dof_index> &array,
+                      const unsigned int                    index)
+      {
+        AssertIndexRange(index, array.size());
+        return array[index];
+      }
+
+      static types::global_dof_index
+      get_array_entry(const std::vector<types::global_dof_index> &array,
+                      const unsigned int                          index)
+      {
+        AssertIndexRange(index, array.size());
+        return array[index];
+      }
+
+      static unsigned int
+      get_array_entry(const int &, const unsigned int)
+      {
+        return 0;
+      }
```

Comment:
I agree. The `int` type is actually a workaround for the case that I do not want to fill a `local_dof_indices` array, but only manipulate the stored data elements. (And it makes no sense to add a second set of functions just for that purpose.) As a result, the `int` type is rather confusing, as it implies we want to do something, rather than simply skip operating on it. I think this might be the case for `std::tuple<>`. Let me try this out.

Reply:
My second commit switches to this more self-explanatory type - I think this is much nicer.